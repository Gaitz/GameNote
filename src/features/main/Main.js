import React from "react"
import styles from "./Main.style.css"
import { useSelector } from "react-redux"

export default function Main() {
  const userInfo = useSelector((state) => state.authentication.user)

  return (
    <>
      <article>
        <h1 className={styles.h1}>Game Note</h1>
        <p>Welcome: {userInfo?.email}</p>
      </article>
    </>
  )
}
