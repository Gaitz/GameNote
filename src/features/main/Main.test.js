import React from "react"
import { render } from "@testing-library/react"
import Main from "./Main"

jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useSelector: jest.fn().mockReturnValue({ email: "test@gmail.com" }),
}))

test("main component snapshot testing", () => {
  const component = render(<Main></Main>)
  let tree = component.container
  expect(tree).toMatchSnapshot()
})
