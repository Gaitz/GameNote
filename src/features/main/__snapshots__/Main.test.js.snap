// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main component snapshot testing 1`] = `
<div>
  <article>
    <h1>
      Game Note
    </h1>
    <p>
      Welcome: 
      test@gmail.com
    </p>
  </article>
</div>
`;
